apps/src/lib/levelbuilder/code-docs-editor/ProgrammingExpressionsTable.jsx

```diff
@@ -0,0 +1,251 @@
+      setCategoriesAvailableForSelect(categoriesForSelect);
+    } else {
+      setCategoriesAvailableForSelect(
+        categoriesForSelect.filter(
+          cat => String(cat.envId) === String(newSelectedEnvironment)
+        )
+      );
+      if (String(selectedCategory.envId) !== newSelectedEnvironment) {
+        setSelectedCategory(DEFAULT_VALUE);
+      }
+    }
+    setCurrentPage(1);
+  };
+
+  const getColumns = () => {
+    return [
+      {
+        property: 'actions',
+        header: {
+          label: 'Actions',
+          props: {
+            style: {width: '10%'}
+          }
+        },
+        cell: {
+          formatters: [actionsCellFormatter]
+        }
+      },
+
+      {
+        property: 'name',
+        header: {
+          label: 'Name'
+        }
+      },
+      {
+        property: 'environmentTitle',
+        header: {
+          label: 'IDE'
+        }
+      },
+      {
+        property: 'categoryName',
+        header: {
+          label: 'Category'
+        }
+      }
+    ];
+  };
+  if (hidden) {
+    return null;
+  }
+  return (
+    <>
+      <select
+        onChange={onEnvironmentSelect}
+        value={selectedEnvironment}
+        style={{marginRight: 7}}
+      >
+        <option value={DEFAULT_VALUE}>All IDEs</option>
+        {programmingEnvironmentsForSelect.map(env => (
+          <option key={env.id} value={env.id}>
+            {env.title || env.name}
+          </option>
+        ))}
+      </select>
+      <select
+        onChange={e => {
+          setSelectedCategory(e.target.value);
+          setCurrentPage(1);
+        }}
+        value={selectedCategory}
+      >
+        <option value={DEFAULT_VALUE}>All Categories</option>
+        {categoriesAvailableForSelect.map(category => (
+          <option key={category.id} value={category.id}>
+            {category.formattedName}
+          </option>
+        ))}
+      </select>
+      {error && <div>{error}</div>}
+      <Table.Provider columns={getColumns()} style={{width: '100%'}}>
+        <Table.Header />
+        <Table.Body rows={programmingExpressions} rowKey="id" />
+      </Table.Provider>
+      <PaginationWrapper
+        totalPages={numPages}
+        currentPage={currentPage}
+        onChangePage={setCurrentPage}
+      />
+      {!!itemToDelete && (
+        <StylizedBaseDialog
+          body={`Are you sure you want to remove ${itemToDelete.name ||
+            itemToDelete.key} and its associated code doc?`}
+          handleConfirmation={() => {
+            destroyExpression(
+              `/programming_expressions/${itemToDelete.id}`,
+              () => {
+                setItemToDelete(null);
+                fetchExpressions(
+                  selectedEnvironment,
+                  selectedCategory,
+                  currentPage
+                );
+              }
+            );
+          }}
+          handleClose={() => setItemToDelete(null)}
+          isOpen
+        />
```

Comment:
(minor) Maybe split this out in a separate function called `renderDeleteConfirmationDialog` or something like that?  Same comment for the other dialog below.

Reply:
I ended up splitting both of these into one function `renderDialogs()`. Is that okay would you rather they be their own functions?

Side note: the render function looks a lot cleaner with the various render functions pulled out :)